### PR TITLE
[Merged by Bors] - feat: add a few analytic function lemmas for future AnalyticManifold use

### PR DESCRIPTION
**Pull request by girving**
This is a few isolated lemmas split out of
https://github.com/leanprover-community/mathlib4/pull/10853.

1. AnalyticAt.comp_of_eq
2. FormalMultilinearSeries.zero_radius
3. constFormalMultilinearSeries_zero

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/Analytic/Basic.lean [hunk 1]
```diff
@@ -182,6 +182,11 @@ theorem constFormalMultilinearSeries_radius {v : F} :
     (by simp [constFormalMultilinearSeries])
 #align formal_multilinear_series.const_formal_multilinear_series_radius FormalMultilinearSeries.constFormalMultilinearSeries_radius
 
+/-- `0` has infinite radius of convergence -/
```
Comment (grunweg) [inline Mathlib/Analysis/Analytic/Basic.lean:185]: Does this generalise to any constant formal series?

File: Mathlib/Analysis/Calculus/FormalMultilinearSeries.lean [hunk 2]
```diff
@@ -362,6 +362,17 @@ theorem constFormalMultilinearSeries_apply [NontriviallyNormedField 𝕜] [Norme
   Nat.casesOn n (fun hn => (hn rfl).elim) (fun _ _ => rfl) hn
 #align const_formal_multilinear_series_apply constFormalMultilinearSeries_apply
 
+@[simp]
```
Comment (grunweg) [inline Mathlib/Analysis/Calculus/FormalMultilinearSeries.lean:365]: Can you generalise this to any `c : F` (or am I misunderstanding how `FormatMultilinearSeries` works)?

**Review by grunweg — COMMENTED** [review]
Thanks for splitting this out. This looks good to me - re-reviewing, I just have one question about generalisation.

File: Mathlib/Analysis/Analytic/Basic.lean (hunk 1 above)
Comment (girving) [reply Mathlib/Analysis/Analytic/Basic.lean:185]: Not really, as this is the 0 of that type, not an injection from the field 0.

File: Mathlib/Analysis/Calculus/FormalMultilinearSeries.lean (hunk 2 above)
Comment (girving) [reply Mathlib/Analysis/Calculus/FormalMultilinearSeries.lean:365]: This is saying the injection from the field commutes with Zero, so I don’t think it sensibly generalizes.

**Comment by girving** [comment]
Are there changes I should make?

**Comment by grunweg** [comment]
No worries, I saw your comments. Right now, I don't have the time to wrap my head around your comments in detail - but what you say sounds reasonable, so let me
maintainer merge

**Comment by sgouezel** [comment]
bors r+
Thanks!

**Pull request closed without merging**